share/availability/discovery/discovery.go

```diff
@@ -151,32 +147,38 @@ func (d *Discovery) Peers(ctx context.Context) ([]peer.ID, error) {
 // TODO: Start advertising only after the reachability is confirmed by AutoNAT
 func (d *Discovery) Advertise(ctx context.Context) {
 	if d.params.AdvertiseInterval == -1 {
+		log.Warn("AdvertiseInterval is set to -1. Skipping advertising...")
 		return
 	}
 
 	timer := time.NewTimer(d.params.AdvertiseInterval)
 	defer timer.Stop()
 	for {
-		ttl, err := d.disc.Advertise(ctx, rendezvousPoint)
+		_, err := d.disc.Advertise(ctx, rendezvousPoint)
 		if err != nil {
-			log.Debugf("Error advertising %s: %s", rendezvousPoint, err.Error())
 			if ctx.Err() != nil {
 				return
 			}
+			log.Debugf("error advertising %s: %s", rendezvousPoint, err.Error())
 
+			errTimer := time.NewTimer(time.Minute)
 			select {
-			case <-timer.C:
+			case <-errTimer.C:
+				errTimer.Stop()
+				if !timer.Stop() {
+					<-timer.C
+				}
 				timer.Reset(d.params.AdvertiseInterval)
```

Comment:
Does it make sense to reset advertiseTimer only after successful advertise?

Reply:
Yes, makes sense